CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+cmake_minimum_required(VERSION 3.15)
+
+file(STRINGS "VERSION" pVersion)
+
+project(
+  g2c
+  VERSION ${pVersion}
+  LANGUAGES C)
+
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+                                               "MinSizeRel" "RelWithDebInfo")
+endif()
+
+if(NOT CMAKE_C_COMPILER_ID MATCHES "^(Intel|GNU|Clang|AppleClang)$")
+  message(WARNING "Compiler not officially supported: ${CMAKE_C_COMPILER_ID}")
+endif()
+
+find_package(Jasper REQUIRED)
+find_package(PNG REQUIRED)
+find_package(ZLIB REQUIRED)
+
+if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_C_FLAGS "-g -O3")
+  set(CMAKE_C_FLAGS_RELEASE "-O3")
+elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_C_FLAGS "-g -O3")
+  set(CMAKE_C_FLAGS_DEBUG "-ggdb -Wall")
+endif()
+
+add_subdirectory(src)
+
+set(lib_name ${PROJECT_NAME})
+set(versioned_lib_name ${PROJECT_NAME}_v${PROJECT_VERSION})
+
+add_library(${lib_name} STATIC ${c_src})
+
+target_compile_definitions(${lib_name} PUBLIC __64BIT__)
+if(JASPER_FOUND)
```

Comment:
This will have to be TRUE, since `Jasper` is `REQUIRED` in L23 `find_package(Jasper REQUIRED)`.
Same for PNG.
So I don't think you need to have the `if() - endif()` for `JASPER` and `PNG`.

Reply:
Building g2c with PNG and/or Japser is in fact optional, so these libraries are not required. I added two cmake options (USE_PNG and USE_Jasper) in https://github.com/NOAA-EMC/NCEPLIBS-g2c/pull/1/commits/551e17df7ac127371b487ad2873404fbdd7b9d7c